engine/src/main/java/nl/inl/blacklab/indexers/config/saxon/XPathFinder.java

```diff
@@ -41,38 +47,49 @@ public class XPathFinder {
     /** Variables to make available from XPath */
     private final Map<String, String> vars = new HashMap<>();
 
-    public XPathFinder(Map<String, String> namespaces, Map<String, String> vars) {
-        this.vars.putAll(vars);
-
-        // Set up namespace aware xpath that will compile xpath expressions
-        xPath = SaxonHelper.newXPathFactory();
-        xPath.setCaching(true); // cache xpaths
+    private final LoadingCache<Map<String, String>, XPathCompiler> compilerCache = Caffeine.newBuilder()
+        .maximumSize(50) // should be large enough for most use cases?
```

Comment:
Should be static, or it won't cache anything? But then we can't access vars, which currently only contains fromInputFile, which is different for every indexed file, so we still wouldn't benefit from caching?